torch_xla/runtime.py

```diff
@@ -44,10 +45,15 @@ def _maybe_select_default_device():
   if torch_xla._found_libtpu and tpu.num_available_chips() > 0:
     logging.warning('libtpu.so and TPU device found. Setting PJRT_DEVICE=TPU.')
     os.environ[xenv.PJRT_DEVICE] = 'TPU'
-  # TODO(wcromar): Detect GPU device
   elif xu.getenv_as(xenv.GPU_NUM_DEVICES, int, 0) > 0:
     logging.warning('GPU_NUM_DEVICES is set. Setting PJRT_DEVICE=CUDA')
     os.environ[xenv.PJRT_DEVICE] = 'CUDA'
+  elif torch.cuda.is_available():
```

Comment:
Out of curiosity, does this work if you use `torch` compiled without CUDA? I think this is what our CI uses.

Reply:
Not sure, is that actually a viable option? As in the current XLA docs assume that you compiled PyTorch with CUDA:

https://github.com/pytorch/xla/blob/master/docs/gpu.md#develop-pytorchxla-on-a-gpu-instance-build-pytorchxla-from-source-with-gpu-support

Should CI/CD be updated to compile with CUDA?